materials/internals/definition.yaml

```diff
@@ -0,0 +1,350 @@
+      color-purple: '#b866ff'
+      color-black: '#1a1a1a'
+      color-white: '#fff'
+  borderRadius:
+    label: Border Radius
+    prefix: border-radius
+    choices:
+      border-radius-1: 1px
+      border-radius-2: 2px
+      border-radius-4: 4px
+      border-radius-6: 6px
+      border-radius-20: 20px
+  shadows:
+    label: Shadows
+    prefix: shadow
+    choices:
+      shadow-1: '0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)'
+      shadow-2: '0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23)'
+      shadow-3: '0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23)'
+      shadow-4: '0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22)'
+      shadow-5: '0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.22)'
+      shadow-6: '0 -1px 2px 0 rgba(0, 0, 0, 0.2)'
+      shadow-7: '0 -1px 1.5px 0 rgba(0, 0, 0, 0.12), 0 1px 1px 0 rgba(0, 0, 0, 0.24)'
+      shadow-8: '0 1px 9.5px 0 rgba(0, 0, 0, 0.12), 0 2px 4px 0 rgba(0, 0, 0, 0.24)'
+      shadow-9: 'inset 0 -1px 3px 0 rgba(0, 0, 0, 0.1), inset 0 1px 1.5px 0 rgba(0, 0, 0, 0.2), inset 0 1px 1.5px 0 rgba(0, 0, 0, 0.25)'
+      shadow-10: '0 0 0 1px rgba(224, 230, 237, 0.5)'
+      shadow-11: '0 1px 0.5px rgba(0, 0, 0, 0.24), 0 -1px 0.75px rgba(0, 0, 0, 0.12)'
+      shadow-12: '0 0 1px rgba(0, 0, 0, 0.25)'
+      shadow-13: '0 0 1px rgba(0, 0, 0, 0.25)'
+      shadow-14: '0 0 0.5px rgba(0, 0, 0, 0.1)'
+      shadow-15: '0 2px 2px rgba(0, 0, 0, 0.24), 0 1px 4.75px rgba(0, 0, 0, 0.12)'
+  constraints:
+    label: Constraints
+    prefix: constraint
+    choices:
+      constraint-scale: 100%
+      constraint-xs: 50px
+      constraint-s: 132px
+      constraint-m: 355px
+      constraint-l: 400px
+      constraint-xl: 768px
+  spacings:
+    label: Spacings
+    prefix: spacing
+    choices:
+      spacing-4: 4px
+      spacing-8: 8px
+      spacing-16: 16px
+      spacing-24: 24px
+      spacing-32: 32px
+  transitions:
+    label: Transitions
+    prefix: transition
+    choices:
+      transition-linear-80ms: '80ms linear'
+      transition-easeinout-150ms: '150ms ease-in-out'
+      transition-standard: 'all 0.2s ease'
+  breakpoints:
+    label: Breakpoints
+    prefix: break-point
+    choices: 
+      break-point-mobile: 768px
+      break-point-desktop: 1024px
+      break-point-biggerdesktop: 1280px
+      break-point-giantdesktop: 1680px
+      break-point-jumbodesktop: 1920px
+
+
+states:
+  hovered:
+    description: 'When the mouse is over the element'
+  # not sure we should actually have this one
+  pristine:
+    description: 'The element was not interacted with yet'
+  disabled:
+    description: 'When the element can not be interacted with'
+  readonly:
+    description: 'When the element can not be modified'
+  warning:
+    description: 'When the element has slightly wrong input'
+  error:
+    description: 'When the element has invalid input'
+
+componentGroups:
+  input:
+    description: 'Input elements'
```

Comment:
I just used the `input` to get us started. Ultimately we identified the following groups. We want to add them as we go instead of defining them all upfront.

- `button`: Clickable elements which trigger an action
- `panel`: Visual grouping
- `dropdown`: Elements which offer more choices in an appearing menu
- `label`: Elements which give meaning to another element and focus the related element when clicked
- `icon`: Icons
- `input`: Elements which get information from the user
- `stamp`: Indicate additional information about an element
- `checkbox`: Elements which allow users to select a subset from a list of options
- `radio`: Elements which allow users to select one option from a list
- `toggle`: Elements which allow users to select a boolean option
- `tag`: Elements which indicate one item of a selection
- `text-headline`: Headline
- `text-subheadline`: Subheadline
- `text-body`: Body Text
- `text-detail`: Detail Text